docs/ConfigurationsViaEnvironmentVariables.md

```diff
@@ -0,0 +1,293 @@
+
+# Transport "xyz"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_NAME="xyz"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_ID="003"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_PORT="9000"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_SECURE="true"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_DESC="This transport will use 9000 as its port"
+```
+
+Where,
+
+* NAMESPACE = `testconfiguration`
+* Each transport is identified uniquely with the "ID" attribute (ex: TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_ID). **Even though "NAME" is also a candidate for the unique identifiable attribute, "ID" is chosen over "NAME" since "ID" has a higher priority.**
+
+#### Example 3 - specifying unique identifiable property via an environment variable / system properties:
+
+Consider the below YAML configuration.
+
+```yaml
+testconfiguration:
+  tenant: tenant
+  transports:
+    transport:
+      -  port: 8000
+         uniqueKey: abc
+         secure: false
+         desc: This transport will use 8000 as its port
+         password: password
+      -  port: 8001
+         uniqueKey: pqr
+         secure: true
+         desc: This transport will use 8001 as its port. Secure - true
+      -  port: 9000
+         uniqueKey: xyz
+         secure: true
+         desc: This transport will use 9000 as its port
+```
+The above configuration can be represented with the below environment variables / system properties:
+
+```bash
+TESTCONFIGURATION_TENANT="tenant"
+
+# Transport "abc"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_UNIQUEKEY="abc"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_PORT="8000"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_SECURE="false"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_DESC="This transport will use 8000 as its port"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_PASSWORD="password"
+
+# Transport "pqr"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_UNIQUEKEY="pqr"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_PORT="8001"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_SECURE="true"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_DESC="This transport will use 8001 as its port. Secure - true"
+
+# Transport "xyz"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_UNIQUEKEY="xyz"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_PORT="9000"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_SECURE="true"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_DESC="This transport will use 9000 as its port"
+```
+
+However, since "ID" and "NAME" attributes are not present, you have to specify your unique identifiable attribute name
+via an environment variable / system property. This variable name is suffixed with the "UNIQUE" keyword.
+
+```bash
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_UNIQUE="UNIQUEKEY"
+```
+
+Where,
+
+* NAMESPACE = `testconfiguration`
+* Each transport is identified uniquely with the "UNIQUEKEY" attribute since it is specified via an environment variable / system property.
+
+## Priority order when building the configuration
```

Comment:
IMO this should be mentioned earlier. Also, we can include recommendations regarding when to use environment variables vs `deployment.yaml` file.